cli/command/stack/kubernetes/convert.go

```diff
@@ -8,13 +8,58 @@ import (
 	"strings"
 
 	"github.com/docker/cli/cli/compose/loader"
-	composeTypes "github.com/docker/cli/cli/compose/types"
+	"github.com/docker/cli/cli/compose/schema"
 	composetypes "github.com/docker/cli/cli/compose/types"
 	"github.com/docker/cli/kubernetes/compose/v1beta1"
 	"github.com/docker/cli/kubernetes/compose/v1beta2"
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// FromCompose converts types.Config (compose) to the specified stack version.
+// It errors out if the version is not supported.
+func FromCompose(version string, stderr io.Writer, name string, cfg *composetypes.Config) (Stack, error) {
+	switch version {
+	case "v1beta1":
+		return v1beta1FromCompose(stderr, name, cfg)
+	case "v1beta2":
+		return v1beta2FromCompose(stderr, name, cfg)
+	default:
+		return Stack{}, errors.Errorf("stack version %s unsupported", version)
+	}
+}
+
+func v1beta1FromCompose(stderr io.Writer, name string, cfg *composetypes.Config) (Stack, error) {
+	cfg.Version = v1beta1.MaxComposeVersion
+	st, err := v1beta2FromCompose(stderr, name, cfg)
```

Comment:
this is confusing; what's beta2 doing in a beta1 conversion?

Reply:
I just moved code, but mainly, `Stack` struct holds both the composefile (`v1beta1` and the `v1beta2` struct) :sweat_smile:

Reply:
cc @silvin-lubecki because I kinda agree we shouldn't need to do that :angel: